pybamm/pybamm_data.py

```diff
@@ -0,0 +1,166 @@
+            "0.1C_discharge_displacement.txt": "sha256:f1329731ead5a82a2be9851cf80e4c6d68dd0774e07aee5361e2af3ab420d7be",
+            "0.5C_discharge_T.txt": "sha256:2140b2f6bd698135d09a25b1f04c271d35a3a02999ace118b10389e01defa2ae",
+            "0.5C_discharge_U.txt": "sha256:9ed8368b2c6149d2a69218e7df6aaade2511c9f7f6fc7932cda153d9a3a10f39",
+            "0.5C_discharge_displacement.txt": "sha256:8098565ff99bc938864797b402f483c1c64a583d6db85d086f39ab0e7b638dd1",
+            "1C_discharge_T.txt": "sha256:97308dfd7f7dd6c434e30f6c00fb6707c43c963855bb0800e0336809d5cc3756",
+            "1C_discharge_U.txt": "sha256:8fc19de45172215d65c56522c224e6fc700ee443db236b814238a829b7a14c3a",
+            "1C_discharge_displacement.txt": "sha256:c2e8617ac48a20921da1b40bbebac479a0a143edf16b12b2e1ff9aaaf1a32ff4",
+            "2C_discharge_T.txt": "sha256:4bd688fb7653539701fe3df61857474b4d54e8b142c84fdc4c8b92b9573fa5d0",
+            "2C_discharge_U.txt": "sha256:7b3c24b5e6df377075002abc2f62bab7c88b27d826812ba5a4c8385a1a12e723",
+            "2C_discharge_displacement.txt": "sha256:2b11513d80827c762325c819a084b87b3a239af7d112f234c9871481760a0013",
+            "stn_2C.txt": "sha256:bb2f90ccfd2cd86ad589287caae13470e554df2f4f47f0f583a5a7e3e6bd9d4c",
+            "stp_2C.txt": "sha256:6fe73b3a18e5fcfb95151dfd7d34c3cbe929792631447ed3ec88c047c9778223",
+            # Drive cycles
+            "UDDS.csv": "sha256:9fe6558c17aad3cc08109186923aeb7459cd3097a381c44e854bf22dd12a5a2a",
+            "US06.csv": "sha256:5909eb2ec7983fae86a050ff3b35a2041d0ab698710a6b0f95d5816e348077ba",
+            "WLTC.csv": "sha256:bb2f95018a44ac1425cb9c787c34721192af502c7385f1358f28e4f75df11fd8",
+            "car_current.csv": "sha256:4305b91b9df073cb048c25dd3fae725e06a94fe200e322e5c08db290d6799e36",
+        }
+        self.registry = pooch.create(
+            path=self.path,
+            base_url=f"https://github.com/pybamm-team/pybamm-data/releases/download/{self.version}",
+            version=self.version,
+            registry=self.files,
+        )
+
+    def get_data(self, filename: str):
+        """
+        Fetches the data file from upstream and stores it in the local cache directory under pybamm directory.
+
+        Parameters
+        ----------
+        filename : str
+            Name of the data file to be fetched from the registry.
+        Returns
+        -------
+        POSIX/Windows Path to the data file.
+        """
+        self.registry.fetch(filename)
+        return pathlib.Path(f"{self.path}/{self.version}/{filename}")
+
+    def show_registry(self, checksum: bool = False):
+        """
+        Prints all the files present in the registry along with their checksums (optional).
+
+        Parameters
+        ----------
+        checksum : bool, optional
+            Prints file registries with checksums if True.
+        Returns
+        -------
+        list, dict (if checksum = True)
+        """
```

Comment:
I am not sure how I feel about a method that returns different data types based on the input argument, do we have a better way to do this?

Reply:
I don't feel there might be a practical application for the return data types, it usually just prints the the registry which is the intention of that method and returns what it prints (to assert it in pytest unit tests).

Reply:
We could create a separate function `show_registry_by_checksum()` that returns a dictionary, while this one can return a list. I agree that there is not much of a practical application for this and I don't think end users need to view the checksums for any of the files (because they have been present inside the PyBaMM repository before this PR).

Another option could be to implement something like `get_data("US06.csv").checksum` that can the checksum for an individual file as a `str`, and move away from this one.

Reply:
I decided to remove the checksum parameter overall as it doesn't have any practical application, rather if a user wants to check the checksum of a particular file they could access the `files` attribute of the `data_loader` class which would give them a `dict` of all files and their checksums. I also updated its execution in the example notebook.